contractcourt/chain_arbitrator.go

```diff
@@ -1056,13 +1056,46 @@ type forceCloseReq struct {
 	closeTx chan *wire.MsgTx
 }
 
+// forceCloseContractOption is used for providing functional options to callers
+// of ForceCloseContract(). We don't export this type directly, but rather the
+// concrete functions implementing this type, so that callers can choose the
+// concrete function that best describes their use case for force closing.
+type forceCloseContractOption func(*ChannelArbitrator)
+
+// Functional option for ForceCloseContract() that signifies that the force
+// close is purely user-initiated.
+func UserInitiatedForceClose(ca *ChannelArbitrator) {
+	// Ignore errors since logging force close info is not a critical part
+	// of the force close flow.
+	_ = ca.log.LogLocalForceCloseInfo(localForceCloseInfo{
+		userInitiated: true,
+	})
+}
+
+// Functional option for ForceCloseContract() that signifies that the force
+// close is due to a link failure. Requires the error message of the link
+// failure to be specified.
+//
+//nolint:revive // forceCloseContractOption doesn't need to be exported
```

Comment:
I don't think there's any harm in exporting `forceCloseContractOption`? I'd rather do that than have a `nolint` directive personally.